### ISSUE TITLE
Ingress controller does not work out of the box when installing the Helm chart and configuring AMBASSADOR_ID

### ISSUE THREAD

**Issue by eevdev** [emissary-ingress/emissary#4691]
**Describe the bug**
I encountered some problems while trying to get Emissary-ingress to act as an `Ingress` controller after installing Emissary-ingress using the Helm chart and configuring `AMBASSADOR_ID`:

1. Helm - generated `IngressClass` is missing the required `getambassador.io/ambassador-id` annotation
If `AMBASSADOR_ID` is set then the annotation `getambassador.io/ambassador-id` should be included in the `IngressClass` generated by the Helm chart, otherwise the `IngressClass` will be ignored ([source](https://github.com/emissary-ingress/emissary/blob/507bf91f95e489d7d5cb437ca7cd1d8f320d4a81/python/ambassador/fetch/ingress.py#L37-L41)).

2. Helm - misleading option to specify `IngressClass` controller value
The chart `IngressClass` template was added in https://github.com/emissary-ingress/emissary/issues/3975 ([PR](https://github.com/emissary-ingress/emissary/pull/4169/files)). Although a part of the suggested solution was to specify the controller class (default is `getambassador.io/ingress-controller`), the solution only touched the `IngressClass` and not other references; this means that Emissary-ingress still requires the controller value to be `getambassador.io/ingress-controller`, so it makes sense to remove the option to configure this value from the chart as changing it would not work.

3. Emissary-ingress - Ingress controller - it should be unnecessary to specify ambassador-id on both `Ingress` and `IngressClass`
If `AMBASSADOR_ID` is set, there's currently a requirement that the `getambassador.io/ambassador-id` annotation must be set on both the `Ingress` and the `IngressClass` (if used instead of the [deprecated `kubernetes.io/ingress.class` annotation](https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation)). Setting the annotation on either `Ingress` or the assigned `IngressClass` should be sufficient.

**To Reproduce**
(I can't remember if it was necessary to restart Emissary-ingress pods after changing `Ingress` and/or `IngressClass` in the steps below)

1. Install Emissary-ingress using the Helm chart and specify `AMBASSADOR_ID` (in my case I have two installations in the same namespace, one for internal traffic and the other for public-facing services)
2. Set up an `Ingress` and assign the generated `IngressClass` to `spec.ingressClassName` (don't add the `getambassador.io/ambassador-id` annotation yet)
3. If the generated `IngressClass` has the `getambassador.io/ambassador-id` annotation then the expected behavior is that the `Ingress` is handled, but the `Ingress` is not handled by Emissary-ingress in this case (Ingress controller requires the annotation on both `Ingress` and `IngressClass`)
4. Add the `getambassador.io/ambassador-id` annotation to the `Ingress`
5. Observe that the `Ingress` is still not handled by Emissary-ingress unless you also have the annotation on the `IngressClass`
6. Add the `getambassador.io/ambassador-id` annotation to the `IngressClass` and observe that Emissary-ingress now picks up the `Ingress`

**Expected behavior**
The `Ingress` should be handled by Emissary-ingress at step 3 (with annotation on `IngressClass` but not on `Ingress`) and after step 4 (with annotation on `Ingress` and with or without the annotation on `IngressClass`).

**Versions (please complete the following information):**
 - Emissary: 3.3.0
 - Helm chart v8.3.0
 - Kubernetes environment: kOps
 - Version: k8s v1.23

**Additional context**
Documentation might also need an update, but since that's a different repository I'll provide a separate PR for that.

**Comment by Bak3y** [comment]
I'm experiencing this exact problem - it'd be very nice for `spec.ingressClassName` to work as expected here - it's what basically every helm chart in existence has hooks into for ingress

**Comment by eevdev** [comment]
I've opened a new PR #5221 to fix the missing `getambassador.io/ambassador-id` annotation for the generated IngressClass.

Note that the [docs for Ingress controller](https://www.getambassador.io/docs/emissary/latest/topics/running/ingress-controller) is a bit out of date/confusing.

> You must create your Ingress resource with the correct ingress.class.
>
> Emissary-ingress will automatically read Ingress resources with the annotation kubernetes.io/ingress.class: ambassador.

> Kubernetes 1.18 introduced the IngressClass resource to the existing networking.k8s.io/v1beta1 API.

The `ingress.class` annotation has been deprecated for a while, 1.18 was EOL 2 years ago, so I think it would be better to provide docs using `spec.ingressClassName` instead.

For an Ingress to be controlled by Emissary Ingress it needs `spec.ingressClassName`, and if you're setting [AMBASSADOR_ID](https://www.getambassador.io/docs/emissary/latest/topics/running/environment#ambassador_id) the annotation `getambassador.io/ambassador-id`. It's possible to set the IngressClass generated by Emissary Ingress as cluster default so you won't have to add this to each of your Ingresses, but you'll need to add the `getambassador.io/ambassador-id` annotation on the Ingress anyway, at least for now.

In my original PR I wanted to remove the need for the annotation on the Ingress, since we can get it from the IngressClass, but I wasn't able to get it to work. Ideally that would also be fixed.

**Comment by eevdev** [comment]
I've scratched out point 1 and 2 from the issue description as 1 has been fixed and 2 isn't very important, leaving just point 3, remove requirement to have AMBASSADOR_ID annotation on both IngressClass and Ingress as long as it's present on one of them.
